### DISCUSSION TITLE
Access the resource ID

### DISCUSSION THREAD

**Discussion by omarbouziane** [laravel/nova-issues#4888] [Q&A]
- Laravel Version: 9.34.0
- Nova Version: 4.15.2
- PHP Version: 8.1.8

Hello, 
I am using nova.pages.edit route in my vue, this route requires two parameters: resource name and resource id, I got the resource name with uriKey() , example: App\nova\User::uriKey(). But I couldn't get the resource Id, I tried  $request->route('resourceId'),  but it didn't work.
Thank you.

**Comment by crynobone** [comment]
Shouldn't you be able to refer to `Laravel\Nova\Http\Controllers\Pages\ResourceUpdateController`?

**Reply by omarbouziane** [reply]
@crynobone 
$request->resourceId returns null.

In dashboard.vue, I have a link to nova.pages.edit. Here is how I want to use the route, in dashboard.vue
`:href="route('nova.pages.edit', ['resourceId', 'resourceName'])"`
  
  I got the resourceName but I still need the resourceId to make it work.
  
  Here is an example how it worked with the index route which requires only resourceName:

```php
Route::get('/dashboard', function () {
    return Inertia::render('Dashboard' ['resourceName', =>User::uriKey()]);
})->middleware(['auth', 'verified'])->name('dashboard');
```

And then in dashboard vue, I passed the data like this:   :href="route('nova.pages.index', $page.props.resourceName)"

**Reply by crynobone** [reply]
I believe you have to figure out how to access `resourceId`. Have no idea how you doing this but if you provide the value to `route()` and assuming you have Ziggy loaded then everything should work.

Nova internally doesn't use `route()` on the front-end (or uses Ziggy) so we can't comment much about it.
